### PR TITLE
Improve release process instruction

### LINKED ISSUE #258: Improve release instruction based on 0.9.0 experience
**Description**

Improve release instruction based on experience with release 0.9.0.

**Reasons**

Instruction should be as clear as possible for people doing the release.

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Improve release process instruction

**Related issue(s)**
Resolves #258

### CODE REVIEW

File: guidelines/releases/release-process.md [hunk 1]
```diff
@@ -4,58 +4,78 @@ This document describes how to create a Kyma release using Prow.
...
-   - for every test
-   - `kyma-docs`
-   - `kyma-integration`
-   - `kyma-gke-integration`
-   - `kyma-gke-upgrade`
-   - `kyma-artifacts`
-   - `kyma-installer`
-
-    >**NOTE:** Read [here](https://github.com/kyma-project/test-infra/blob/master/docs/prow/release-jobs.md) how to define a release job for a component.
+        > **Note:** Remember about modifying `presets` array. For example, for 0.9 release the preset is the following:
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
        > **Note:** Remember to modify the `presets` array. This is an example of the 0.9 release `preset`:
```

File: guidelines/releases/release-process.md [hunk 2]
```diff
@@ -4,58 +4,78 @@ This document describes how to create a Kyma release using Prow.
...
+        ```
+        - labels:
+            preset-target-commit-0.9: "true"
+            env:
+            - name: RELEASE_TARGET_COMMIT
+                value: release-0.9
+        ```
 
-2. Ensure that tests for release jobs exist. Release tests usually iterate through all release versions and run tests for them.
+1. Ensure that tests for release jobs exist. Release tests usually iterate through all release versions and run tests for them.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
1. Ensure that tests for the release jobs exist. Release tests usually iterate through all release versions and run tests for them.
```

File: guidelines/releases/release-process.md [hunk 3]
```diff
@@ -4,58 +4,78 @@ This document describes how to create a Kyma release using Prow.
...
+          - pre-rel09-kyma-artifacts
+          - pre-rel09-kyma-installer
+          - pre-rel09-kyma-gke-upgrade
+          - pre-rel09-kyma-gke-central-connector
     ```
 
-## Release
+### Remove previous release jobs
+
+After adding new release jobs, remove all previous ones. For example, while doing 0.9 release, add `pre-rel09` jobs and remove all `pre-rel06` jobs. Make sure that there are only defined jobs with `pre-rel07`, `pre-rel08` and `pre-rel09` prefixes.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: Hmm, so you don't remove ALL the previous jobs as such but you leave those from three releases back, right?

File: guidelines/releases/release-process.md [hunk 4]
```diff
@@ -4,58 +4,78 @@ This document describes how to create a Kyma release using Prow.
...
+
+## Do release
 
 > **NOTE:** Currently, you are most likely required to push to the `release-x.y` branch more than once. Make sure that a user with **admin** role is present in the `kyma` repository. 
-> Before every release you must bump the version in `prow/RELEASE_VERSION` directory of the `test-infra` repository. Make sure it is approved by a member of wg-prow. 
+
+Before doing the release, make sure there is no mismatch between source code and YAML files. All components are rebuilt from source code, which means all Helm charts need to be up to date.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
Before a release, make sure there is no mismatch between source code and `.yaml` files. All components are rebuilt from source code, which requires all Helm charts to be up to date.
```

File: guidelines/releases/release-process.md [hunk 5]
```diff
@@ -67,28 +87,35 @@ Follow these steps to create a release:
...
+    echo -n {RELEASE_VERSION} > prow/RELEASE_VERSION
+    ```
 
-3. Push the branch to the `test-infra`  repository.
+3. Push the branch to the `test-infra` repository.
 
-4. Create a PR to `test-infra/release-x.y`. This triggers the pre-release job for watch-pods.
+4. Create a PR to `test-infra/release-x.y`. This triggers the pre-release job for `watch-pods`. 
+
+    > **NOTE:** If you want to just trigger `watch-pods` build without introducing any changes, edit any file within `test-infra` repository and create a pull request. You don't need to merge it - the job will be triggered anyway. After successful `watch-pods` image build, you can close the pull request.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
    > **NOTE:** To trigger the `watch-pods` build without introducing any changes, edit any file within the `test-infra` repository and create a pull request. You don't need to merge it as the job will be triggered anyway. After a successful `watch-pods` image build, you can close the pull request.
```

File: guidelines/releases/release-process.md [hunk 6]
```diff
@@ -67,28 +87,35 @@ Follow these steps to create a release:
...
+
+5. Prepare and merge pull request to `master` with updating RELEASE_VERSION for next minor release candidate. For example, if `RELEASE_VERSION` on `master` branch is set to `0.9.2`, then change the version to `1.0.0-rc1`.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: What do you mean by preparing the pull request? Is the `RELEASE_VERSION` a file? Would you agree with the below?
```suggestion
5. Update the `RELEASE_VERSION`file with the name of the next minor release candidate and merge the pull request to `master`. For example, if the `RELEASE_VERSION` on the `master` branch is set to `0.9.2`, then change the version to `1.0.0-rc1`.
```

File: guidelines/releases/release-process.md [hunk 7]
```diff
@@ -67,28 +87,35 @@ Follow these steps to create a release:
...
-    git checkout -b $RELEASE_NAME upstream/master
-    git push upstream $RELEASE_NAME
+    git checkout -b release-{release_version} upstream/master
+    git push upstream release-{release_version}
     ```
 
-2. Create a PR for the `kyma` release branch.
-    ![PullRequest](../../assets/release-PR.png)
-    This triggers all jobs for components.
+    > **NOTE:** Pushing anything to release branch (and also pushing rebased branch) results in creating new Github release.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: I'm not sure what you meant, could you rephrase?
```suggestion
    > **NOTE:** If you push anything to the release branch, including the rebased branch, a new Github release is created.
```

File: guidelines/releases/release-process.md [hunk 8]
```diff
@@ -102,46 +129,39 @@ Follow these steps to create a release:
...
-                  labels:
-                    name: kyma-installer
-                spec:
-                  serviceAccountName: kyma-installer
-                  containers:
-                  - name: kyma-installer-container
-                    image: eu.gcr.io/kyma-project/develop/installer:55bc6038
-                    imagePullPolicy: IfNotPresent
-             ```
+        - In `installation/resources/installer.yaml` change from `eu.gcr.io/kyma-project/develop/installer:{image_tag}` to `eu.gcr.io/kyma-project/kyma-installer:{release_version}`
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
        - In `installation/resources/installer.yaml` replace`eu.gcr.io/kyma-project/develop/installer:{image_tag}` with `eu.gcr.io/kyma-project/kyma-installer:{release_version}`
```

File: guidelines/releases/release-process.md [hunk 9]
```diff
@@ -102,46 +129,39 @@ Follow these steps to create a release:
...
+    
+    3. In `resources/core/values.yaml` file, replace the `clusterDocsTopicsVersion` value to your release branch name. For example, for 0.9.1 release, find the following section:
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
    3. In  the `resources/core/values.yaml` file, replace the `clusterDocsTopicsVersion` value with your release branch name. For example, for the 0.9.1 release, find the following section:
```

File: guidelines/releases/release-process.md [hunk 10]
```diff
@@ -102,46 +129,39 @@ Follow these steps to create a release:
...
+
+        ```yaml
+        docs:
+            # (...) - truncated
+            clusterDocsTopicsVersion: master
+        ```
+
+        Replace the value `clusterDocsTopicsVersion` to the following:
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
        And replace the `clusterDocsTopicsVersion` value with the following:
```

File: guidelines/releases/release-process.md [hunk 11]
```diff
@@ -172,13 +192,15 @@ Follow these steps to create a release:
 7. After all checks pass, merge the PR.
     > **NOTE:** To merge the PR to the release branch, you must receive approvals from all teams.
 
-8. Merging the PR to the release branch runs the postsubmit job that creates a GitHub release.
+8. Merging the PR to the release branch runs the postsubmit job that creates a GitHub release. This job also triggers documentation update on the official Kyma website.
+
+9. Update `RELEASE_VERSION` to the next patch RC1 version on the release branch. Do it immediately after the release, otherwise any PR to a release branch overrides the previously published Docker images.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: I don't get the next patch RC1 version.

File: guidelines/releases/release-process.md [hunk 12]
```diff
@@ -172,13 +192,15 @@ Follow these steps to create a release:
...
 
-9. Update `RELEASE_VERSION` to the next version both on the `master` and release branches. Do it immediately after the release, otherwise any PR to a release branch overrides the previously published Docker images.
+    For example, if `RELEASE_VERSION` on release branch contains `0.9.1`, change it to `0.9.2-rc1`.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
    For example, if the `RELEASE_VERSION` on the release branch is `0.9.1`, change it to `0.9.2-rc1`.
```

File: guidelines/releases/release-process.md [hunk 13]
```diff
@@ -172,13 +192,15 @@ Follow these steps to create a release:
...
 
 10.  Validate the `yaml` and changelog files generated under [releases](https://github.com/kyma-project/kyma/releases).
-11. Update the release content manually with the instruction on how to install the latest Kyma release.  
-    Currently this means to grab the instructions from the previous release and change the version number in them. If contributors want you to change something in them they would address you directly.
 
+11. Update the release content manually with the links how to install the latest Kyma release.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
11. Update the release content manually with links to the instruction on how to install the latest Kyma release.  
```

File: guidelines/releases/release-process.md [hunk 14]
```diff
@@ -172,13 +192,15 @@ Follow these steps to create a release:
...
 
-9. Update `RELEASE_VERSION` to the next version both on the `master` and release branches. Do it immediately after the release, otherwise any PR to a release branch overrides the previously published Docker images.
+    For example, if `RELEASE_VERSION` on release branch contains `0.9.1`, change it to `0.9.2-rc1`.
 
 10.  Validate the `yaml` and changelog files generated under [releases](https://github.com/kyma-project/kyma/releases).
-11. Update the release content manually with the instruction on how to install the latest Kyma release.  
-    Currently this means to grab the instructions from the previous release and change the version number in them. If contributors want you to change something in them they would address you directly.
 
+11. Update the release content manually with the links how to install the latest Kyma release.  
+    Currently this means to grab the links from the previous release and update the version number in URLs. If contributors want you to change something in them they would address you directly.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: Do contributors want to change something in the links or in the instruction?
```suggestion
    Currently, this means to grab the links from the previous release and update the version number in URLs. If contributors want you to change something in the instruction, they would address you directly.
```

File: guidelines/releases/release-process.md (hunk 3 above)
Comment (pkosiec) [reply guidelines/releases/release-process.md]: Yes, exactly

File: guidelines/releases/release-process.md (hunk 11 above)
Comment (pkosiec) [reply guidelines/releases/release-process.md]: yes, it always refer to a file.

File: guidelines/releases/release-process.md [hunk 15]
```diff
@@ -4,58 +4,78 @@ This document describes how to create a Kyma release using Prow.
...
-   - for every tool  
-     Exceptions:
+Preparation consists of defining new release jobs and removing old ones.
+
+### Define release jobs
+
+Define the release jobs on the `master` branch in the `test-infra` repository. Since every job name has to be unique, prefix it with `pre-rel{XY}`. To learn how to define a release job for a component, read the following [document](https://github.com/kyma-project/test-infra/blob/master/docs/prow/release-jobs.md). To find all release jobs for the 0.9 release, look for job names with the `pre-rel09` prefix.
+
+1. In `test-infra` repository navigate to `prow/jobs` directory.
+1. Define release jobs in the `test-infra` repository in the following files:
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply guidelines/releases/release-process.md]: What about now? 🙂

File: guidelines/releases/release-process.md (hunk 3 above)
Comment (mmitoraj) [reply guidelines/releases/release-process.md]:
```suggestion
After adding new release jobs, remove the old ones. Remember to leave jobs for three latest releases. For example, during the preparation for the 0.9 release, add `pre-rel09` jobs and remove all `pre-rel06` jobs. Make sure that the only defined jobs are those with `pre-rel07`, `pre-rel08`, and `pre-rel09` prefixes.
```

File: guidelines/releases/release-process.md [hunk 16]
```diff
@@ -67,28 +88,35 @@ Follow these steps to create a release:
...
+    echo -n {RELEASE_VERSION} > prow/RELEASE_VERSION
+    ```
+
+3. Push the branch to the `test-infra` repository.
 
-3. Push the branch to the `test-infra`  repository.
+4. Create a PR to `test-infra/release-x.y`. This triggers the pre-release job for `watch-pods`. 
 
-4. Create a PR to `test-infra/release-x.y`. This triggers the pre-release job for watch-pods.
+    > **NOTE:** To trigger the `watch-pods` build without introducing any changes, edit any file within the `test-infra` repository and create a pull request. You don't need to merge it as the job will be triggered anyway. After a successful `watch-pods` image build, you can close the pull request.
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]:
```suggestion
    > **NOTE:** To trigger the `watch-pods` build without introducing any changes, edit any file within the `test-infra` repository and create a pull request. You don't need to merge it as the job is triggered anyway. After a successful `watch-pods` image build, close the pull request.
```

File: guidelines/releases/release-process.md [hunk 17]
```diff
@@ -102,46 +130,39 @@ Follow these steps to create a release:
...
         ```
 
         > **NOTE:** Replace only `develop/` so `develop/tests` becomes `tests/`.
 
-        Every component image is published with a version defined in the `RELEASE_VERSION` file stored in the `test-infra` repository on the given release branch.
+        Every component image is published with a version defined in the `RELEASE_VERSION` file stored in the `test-infra` repository on the given release branch. Test scripts for integration jobs like GKE Integration or GKE Upgrade are also loaded from the `test-infra` release branch.
 
     2. Check all `yaml` files for the following references:  
-        `image: eu.gcr.io/kyma-project/develop/{{IMAGE_NAME}}:{{SOME_SHA}}`
+        `image: eu.gcr.io/kyma-project/develop/{IMAGE_NAME}:{SOME_SHA}`
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: Add some additional space here as the below renders in the same line

File: guidelines/releases/release-process.md [hunk 18]
```diff
@@ -102,46 +130,39 @@ Follow these steps to create a release:
...
-        version: {rel_version}
+        version: {release_version}
         ```
 
         > **NOTE:** Replace only `develop/` so `develop/tests` becomes `tests/`.
 
-        Every component image is published with a version defined in the `RELEASE_VERSION` file stored in the `test-infra` repository on the given release branch.
+        Every component image is published with a version defined in the `RELEASE_VERSION` file stored in the `test-infra` repository on the given release branch. Test scripts for integration jobs like GKE Integration or GKE Upgrade are also loaded from the `test-infra` release branch.
 
     2. Check all `yaml` files for the following references:
```
Comment (mmitoraj) [inline guidelines/releases/release-process.md]: So actually you replace `image: eu.gcr.io/kyma-project/develop/{IMAGE_NAME}:{SOME_SHA}` with `image: eu.gcr.io/kyma-project/{IMAGE_NAME}:{release_version}` in `installation/resources/installer-local.yaml` and In `eu.gcr.io/kyma-project/develop/installer:{image_tag}` with `eu.gcr.io/kyma-project/kyma-installer:{release_version}` in `installation/resources/installer.yaml`? Maybe we make it clearer by making two seperate points of that (ii and iii)?

File: guidelines/releases/release-process.md (hunk 11 above)
Comment (mmitoraj) [reply guidelines/releases/release-process.md]: so
```suggestion
9. Update the `RELEASE_VERSION` file to contain the next patch RC1 version on the release branch. Do it immediately after the release, otherwise, any PR to a release branch overrides the previously published Docker images.
```

File: guidelines/releases/release-process.md (hunk 18 above)
Comment (pkosiec) [reply guidelines/releases/release-process.md]: What about now? 😄
